crates/nu-command/src/filters/drop/nth.rs

```diff
@@ -102,108 +99,137 @@ impl Command for DropNth {
         input: PipelineData,
     ) -> Result<PipelineData, ShellError> {
         let head = call.head;
-        let metadata = input.metadata();
-        let number_or_range = extract_int_or_range(engine_state, stack, call)?;
-
-        let rows = match number_or_range.item {
-            Either::Left(row_number) => {
-                let and_rows: Vec<Spanned<i64>> = call.rest(engine_state, stack, 1)?;
-                let mut rows: Vec<_> = and_rows.into_iter().map(|x| x.item as usize).collect();
-                rows.push(row_number as usize);
-                rows.sort_unstable();
-                rows
-            }
-            Either::Right(Range::FloatRange(_)) => {
-                return Err(ShellError::UnsupportedInput {
-                    msg: "float range".into(),
-                    input: "value originates from here".into(),
-                    msg_span: head,
-                    input_span: number_or_range.span,
-                });
-            }
-            Either::Right(Range::IntRange(range)) => {
-                // check for negative range inputs, e.g., (2..-5)
-                let end_negative = match range.end() {
-                    Bound::Included(end) | Bound::Excluded(end) => end < 0,
-                    Bound::Unbounded => false,
-                };
-                if range.start().is_negative() || end_negative {
+        let metadata = input.metadata().as_ref().cloned();
```

Comment:
this clone isn't necessary
```suggestion
        let metadata = input.metadata();
```